pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -37,7 +37,13 @@ def dtype(self, new_dtype: Union[str, torch.dtype]):
 
     @property
     def device(self) -> Union[str, torch.device]:
-        return self._device
+        device = self._device
+
+        # make this more explicit to always include the index
+        if device.type == 'cuda' and device.index is None:
+            return torch.device(f'cuda:{torch.cuda.current_device()}')
```

Comment:
passing the index via string vs. additional argument does not make a difference, right?

Reply:
No, that's the same.

> maybe there is a reason why PyTorch does not set the index explicitly in case of index=None?

I was thinking the same, but when you do the following:
```python
import torch
a = torch.rand(1, 2, 3)
a = a.cuda()
print(a.device)
```

you get a cuda device with an index, so this part of the parsing logic must be part of the tensor class and not the module class there.